src/components/views/AgreementForm.js

```diff
@@ -217,6 +219,9 @@ class AgreementForm extends React.Component {
 export default stripesFinalForm({
   initialValuesEqual: (a, b) => isEqual(a, b),
   keepDirtyOnReinitialize: true,
+  subscription: {
```

Comment:
@aditya-matukumalli, is this necessary if he's using a `FormSpy`. It feels like we're subscribing to the same thing (`values`) twice. If that's the case, it'd be nice if we removed one of them for optimization.